app/services/reports/injection_errors.rb

```diff
@@ -70,6 +70,15 @@ def where_conditions
       ['ia.succeeded = false', date_range_clause].compact.join(' AND ')
     end
 
+    def when_cond(error_string, error_type)
+      "WHEN obj->>'error' ILIKE '#{error_string}' THEN '#{error_type}'"
```

Comment:
could use substitution:
```
template = "WHEN obj->>'error' ILIKE %{error_message} THEN %{error_type}"
template % { error_message: error_string. error_type: error_type }
```
Other examples:
https://blog.revathskumar.com/2013/01/ruby-multiple-string-substitution-in-string-template.html